src/components/Tooltip.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { createPortal } from "react-dom";
+
+interface TooltipProps {
+    children: React.ReactElement;
+    offset?: { x: number; y: number };
+    content: React.ReactElement;
+  }
+
+const Tooltip: React.FC<TooltipProps> = ({
+    children,
+    offset = { x: 0, y: 0 },
+    content,
+  }: {
+    children: React.ReactElement;
```

Comment:
we are destructuring / declaring types twice here. Add classname in props and do the destructuring and defaults once.